IframeMediaPause/iframe_media_pausing.md

```diff
@@ -0,0 +1,199 @@
+
+In the past, the ["execution-while-not-rendered" and "execution-while-out-of-viewport"](https://wicg.github.io/page-lifecycle/#feature-policies) permission policies have been proposed as additions to the Page Lifecycle draft specification. However, these policies freeze all iframe JavaScript execution when not rendered, which is not desirable for the featured use case. Moreover, this proposal has not been adopted or standardized.
+
+## Interoperability with other Web API's
+
+Given that there exists many ways for a website to render audio in the broader web platform, this proposal has points of contact with many API's. To be more specific, there are two scenarios where this interaction might happen:
+- Scenario 1: When the iframe is not rendered and it attempts to play audio; and
+- Scenario 2: When the iframe is already playing audio and stops rendering. Once rendering again, to resume playback, we recommend that websites wait for a new user gesture to do so.
+
+For the first scenario, the APIs should behave as if they didn't have permission in the first place to play audio. Like when the [`autoplay` permission policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Permissions-Policy/autoplay) is set to `'none'` for an iframe. Regarding the second scenario, we propose that the API's behave as if the user had paused the media playback.
+
+The following subsections covers how this proposal could interact with Web APIs that render audio.
+
+### HTMLMediaElements
+
+HTMLMediaElement media playback is started and paused, respectively, with the [`play`](https://html.spec.whatwg.org/multipage/media.html#dom-media-play) and [`pause`](https://html.spec.whatwg.org/multipage/media.html#dom-media-pause) methods. For scenario 1, the media element shouldn't be [allowed to play](https://html.spec.whatwg.org/multipage/media.html#allowed-to-play) and `play` should return a promise rejected with `"NotAllowedError"`. In this case, the website could easily handle this like shown below.
+
+```js
+let startPlayPromise = videoElem.play();
+
+if (startPlayPromise !== undefined) {
+  startPlayPromise
+    .then(() => {
+      // Start whatever you need to do only after playback
+      // has begun.
+    })
+    .catch((error) => {
+      if (error.name === "NotAllowedError") {
+        showPlayButton(videoElem);
+      } else {
+        // Handle a load or playback error
+      }
+    });
+}
+```
+<*Snippet extracted from [MDN](https://developer.mozilla.org/en-US/docs/Web/Media/Autoplay_guide)*>
+
+For the scenario 2, when the iframe is not rendered anymore, the media should be [paused](https://html.spec.whatwg.org/multipage/media.html#dom-media-pause) and a [`pause`](https://html.spec.whatwg.org/multipage/media.html#event-media-pause) event should be dispatched.  
```

Comment:
Calling the `pause()` method or even running the [internal pause steps](https://html.spec.whatwg.org/multipage/media.html#internal-pause-steps) involve more than just firing the `pause` event. Ex. the `pause` attribute is set and the `timeupdate` event is fired. Could you clarify if this is equivalent to calling `pause()` or something different?

Reply:
Yes. Our intention is for it to be equivalent to calling `pause()`. That's why I wrote that both things should happen: the media should be paused, and the pause event would be dispatched too.

Reply:
Would recommend writing this down more explicitly, like "the user agent must run the same steps as it would if the `pause()` method was invoked on the media element" and linking to the relevant part of the spec. With that, no need to call out firing the `pause` event explicitly.

Similar suggestion for `play()`.

Reply:
Ok. I am adding these suggestions in this follow-up PR: https://github.com/MicrosoftEdge/MSEdgeExplainers/pull/771